src/control/cmd/daos_server/storage_nvme.go

```diff
@@ -0,0 +1,250 @@
+}
+
+func validateVFIOSetting(targetUser string, reqDisableVFIO bool, iommuEnabled bool) error {
+	if targetUser != "root" {
+		if reqDisableVFIO {
+			return storage.FaultBdevNonRootVFIODisable
+		}
+		if !iommuEnabled {
+			return storage.FaultBdevNoIOMMU
+		}
+	}
+	return nil
+}
+
+func updatePrepReqParams(log logging.Logger, iommuEnabled bool, req *storage.BdevPrepareRequest) error {
+	targetUser, err := getTargetUser(req.TargetUser)
+	if err != nil {
+		return err
+	}
+	// Update target user parameter in request.
+	req.TargetUser = targetUser
+
+	if err := validateVFIOSetting(targetUser, req.DisableVFIO, iommuEnabled); err != nil {
+		return err
+	}
+
+	switch {
+	case req.DisableVFIO:
+		log.Info("VMD not enabled because VFIO disabled in command options")
+	case !iommuEnabled:
+		log.Info("VMD not enabled because IOMMU disabled on platform")
+	default:
+		// If none of the cases above match, set enable VMD flag in request.
+		req.EnableVMD = true
+	}
+
+	return nil
+}
+
+type prepareDrivesCmd struct {
+	cmdutil.LogCmd  `json:"-"`
+	helperLogCmd    `json:"-"`
+	iommuCheckerCmd `json:"-"`
+
+	PCIBlockList string `long:"pci-block-list" description:"Whitespace separated list of PCI devices (by address) to be ignored when unbinding devices from Kernel driver to be used with SPDK (default is no PCI devices)"`
+	NrHugepages  int    `short:"p" long:"hugepages" description:"Number of hugepages to allocate for use by SPDK (default 1024)"`
+	TargetUser   string `short:"u" long:"target-user" description:"User that will own hugepage mountpoint directory and vfio groups."`
+	DisableVFIO  bool   `long:"disable-vfio" description:"Force SPDK to use the UIO driver for NVMe device access"`
+	Args         struct {
+		PCIAllowList string `positional-arg-name:"pci-allow-list" description:"Whitespace separated list of PCI devices (by address) to be unbound from Kernel driver and used with SPDK (default is all PCI devices)"`
+	} `positional-args:"yes"`
+}
+
+func (cmd *prepareDrivesCmd) WithDisableVFIO(b bool) *prepareDrivesCmd {
+	cmd.DisableVFIO = b
+	return cmd
+}
+
+func (cmd *prepareDrivesCmd) WithTargetUser(u string) *prepareDrivesCmd {
+	cmd.TargetUser = u
+	return cmd
+}
+
+func (cmd *prepareDrivesCmd) WithPCIAllowList(al string) *prepareDrivesCmd {
+	cmd.Args.PCIAllowList = al
+	return cmd
+}
+
+func (cmd *prepareDrivesCmd) prepareNVMe(backendCall nvmePrepareResetFn, iommuEnabled bool) error {
+	cmd.Info("Prepare locally-attached NVMe storage...")
+
+	req := storage.BdevPrepareRequest{
+		HugePageCount: cmd.NrHugepages,
+		TargetUser:    cmd.TargetUser,
+		PCIAllowList:  cmd.Args.PCIAllowList,
+		PCIBlockList:  cmd.PCIBlockList,
+		DisableVFIO:   cmd.DisableVFIO,
+	}
+
+	if err := updatePrepReqParams(cmd.Logger, iommuEnabled, &req); err != nil {
+		return errors.Wrap(err, "evaluating vmd capability on platform")
+	}
+
+	cmd.Debugf("nvme prepare request parameters: %+v", req)
+
+	// Configure NVMe device access.
+	_, err := backendCall(req)
```

Comment:
minor - could backendCall be renamed to something like prepare or prepareBackend?